third_party/bigframes_vendored/pandas/core/computation/parsing.py

```diff
@@ -0,0 +1,196 @@
+# Contains code from https://github.com/pandas-dev/pandas/blob/main/pandas/core/computation/parsing.py
+"""
+:func:`~pandas.eval` source string parsing functions
+"""
+from __future__ import annotations
+
+from io import StringIO
+from keyword import iskeyword
+import token
+import tokenize
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Hashable, Iterator
+
+# A token value Python's tokenizer probably will never use.
+BACKTICK_QUOTED_STRING = 100
+
+
+def create_valid_python_identifier(name: str) -> str:
```

Comment:
Aside: I wonder if we should use this function when creating column names? Could make them more interpretable.

Reply:
Could borrow this approach with sql special characters